### PR TITLE
Add mergeable_state and make PR body nullable.

### LINKED ISSUE #245: Pull request body can be null
```
ParseError "Error in $.body: failed to parse field body: expected Text, encountered Null"
```

Pull requests can have no message in the body. In this case, the body text should be empty.

This applies to the currently released as well as the git master version.

### LINKED ISSUE #246: Add more fields (e.g. mergeable_state)
I want to see the `mergeable_state` from the pull request API. There are also a lot of other missing fields. Perhaps there is some IDL or something that you could generate most of this from?

### PR DESCRIPTION
**Pull request by iphydf**
Fixes #245.
Fixes #246.

### CODE REVIEW

File: src/GitHub/Data/PullRequests.hs [hunk 1]
```diff
@@ -77,6 +77,7 @@ data PullRequest = PullRequest
     , pullRequestCommits        :: !Count
     , pullRequestMerged         :: !Bool
     , pullRequestMergeable      :: !(Maybe Bool)
+    , pullRequestMergeableState :: !Text
```
Comment (phadej) [inline src/GitHub/Data/PullRequests.hs]: Create an auxiliary `data` type for this.

File: src/GitHub/Data/PullRequests.hs [hunk 2]
```diff
@@ -216,7 +217,7 @@ instance FromJSON PullRequest where
         <*> o .: "number"
         <*> o .: "html_url"
         <*> o .: "updated_at"
-        <*> o .: "body"
+        <*> o .:? "body" .!= "" -- TODO: no body is treated as empty
```
Comment (phadej) [inline src/GitHub/Data/PullRequests.hs]: I'd rather change a field to have a `Maybe Text` type.

File: src/GitHub/Data/PullRequests.hs (hunk 1 above)
Comment (iphydf) [inline src/GitHub/Data/PullRequests.hs]: Done.

File: src/GitHub/Data/PullRequests.hs (hunk 2 above)
Comment (iphydf) [inline src/GitHub/Data/PullRequests.hs]: Done.

**Pull request merged**